### DISCUSSION TITLE
Feature Request: Support FILE_SIZE_BYTES option in partitioned writes

### DISCUSSION THREAD

**Discussion by mothukur** [duckdb/duckdb#13058] [Ideas]
At my firm, I am working on a project to use DuckDB as a compute engine with the Apache Iceberg table format. I am able to implement most of the features. But one thing I encountered is that I cannot use the FILE_SIZE_BYTES option along with partitioned writes (using PARTITION_BY). In the absence of this, I have to create the required parquet files in two steps: in the first step, writing data using the PARTITION_BY clause, and then, for each partition, writing the data again using the FILE_SIZE_BYTES option. It would be great if both could be done in one single write operation, which would reduce the overall write time. Another thing I have observed is the inability to use the ORDER BY clause with FILE_SIZE_BYTES. Adding this feature also helps with writing data only once.

**Comment by arthurgailes** [comment]
bumping this. arrow has it natively through its row size parameter
